### ISSUE TITLE
[a11y] Talkback does not mention "Show all" as a button in the Recently visited section

### ISSUE THREAD

**Issue by delia-pop** [mozilla-mobile/fenix#27130]
### Steps to reproduce

Make sure the local is on US.

Steps to reproduce
1. Turn TalkBack (or the available screen reader) on.
2. Launch Fenix and open a webpage.
3. Bookmark the webpage.
4. Return to homepage.
5. Tap on the "Show all" button from the "Recently Saved" section and notice what Talkback announces.

### Expected behaviour

When tapping on the "Show all" button, the Talkback should announce "Recently visited Show all past explorations **button** ".

### Actual behaviour

When tapping on the "Show all" button, the Talkback only announces "Recently visited Show all past explorations".

### Device name

Google Pixel 6 

### Android version

Android 13

### Firefox release type

Firefox Nightly

### Firefox version

Nightly 107.0a1 from 09/23

### Additional information

![1663937227077](https://user-images.githubusercontent.com/89388888/191967202-b41ff5fa-f7b3-4dfa-80d1-2d00591f71e1.JPEG)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21648)

**Comment by SoftVision-LorandJanos** [comment]
Still reproducible with the latest Beta 109.0b3 build.
Device used: Oppo Find X5 (Android 12).

**Comment by cpeterso** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1809815

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
